### PR TITLE
Mobile/layout

### LINKED ISSUE #480: Layout dimensions
Considering sizing on tablet and mobile. Consult with UX as required.

Investigate using CSS media queries - take height into account too.

Desired behaviour:

|                  | Mobile Landscape | Mobile Portrait   | Tablet Landscape | Tablet Portrait   |
|------------------|------------------|-------------------|------------------|-------------------|
| **Navigator**        | Hidden           | Hidden            | Shown | Shown             |
| **Period selection** | Dropdown         | Dropdown          | Tabs             | Tabs          |
| **Clear all button** | Shown            | Shown             | Hidden           | Hidden             |

### PR DESCRIPTION
**Pull request by rjmcneill**
Relates to #480 

Current configuration for this should match up with the UX designs.
Tested with iPhone 6 and iPad.

Legend and crosshair also hidden for all touch screens

### CODE REVIEW

File: src/assets/styles/head.less [hunk 1]
```diff
@@ -16,6 +16,25 @@
     }
 
     #period-selector {
+      // Phone portrait
+      @media (max-height: @screen-xs-max) and (max-width: @screen-xs-max) and
```
Comment (djmiley) [inline src/assets/styles/head.less]: Is there any way to declare these separately as variables and re-use them, as they are repeated a few times?

**Comment by djmiley** [comment]
Looks quite concise, I like it.

File: src/assets/styles/head.less (hunk 1 above)
Comment (jleft) [inline src/assets/styles/head.less]: In Less - yes! [Mixins](http://lesscss.org/features/#mixins-feature) are very powerful and can help reduce duplication, this looks like a good time to make use of them.

For some inspiration...

E.g. This Less:

```less
@screen-xs-max: 100px;

.phone-portrait-display(@display) {
  @media (max-height: @screen-xs-max) and (max-width: @screen-xs-max) and (orientation: portrait) { 
    display: @display;
  }
}

.some-class {
  .phone-portrait-display(none) ;
}
```

Would compile to this CSS:

```css
@media (max-height: 100px) and (max-width: 100px) and (orientation: portrait) {
  .some-class {
    display: none;
  }
}
```
---

Another example. This Less:

```less
@screen-xs-max: 100px;

.phone-display(@display, @orientation) {
  @media (max-height: @screen-xs-max) and (max-width: @screen-xs-max) and (orientation: @orientation) { 
    display: @display;
  }
}

.some-class {
  .phone-display(block, landscape) ;
}
```

Would compile to this CSS:

```css
@media (max-height: 100px) and (max-width: 100px) and (orientation: landscape) {
  .some-class {
    display: block;
  }
}

```

**Comment by jleft** [comment]
> Looks quite concise, I like it.

I agree - I'm surprised at how concise it is - nice work :+1:

I'd still be interested to see if it can be simplified with the use of mixins, as described above.

File: src/assets/styles/head.less (hunk 1 above)
Comment (rjmcneill) [inline src/assets/styles/head.less]: Thanks, I was actually just trying to figure out how to implement this!
I'll try get that tidied up now :)

File: src/assets/styles/variables.less [hunk 2]
```diff
@@ -160,3 +160,62 @@
 @spinner-size: 50px;
 @spinner-thickness: 7px;
 @spinner-animation-duration: 0.6s;
+
+// Device Configurations
+// If the period dropdown is shown, the period tab and clear all should be hidden
+
+.phone-portrait-display(@display) {
```
Comment (jleft) [inline src/assets/styles/variables.less]: Please can we move the mixins into a new file in the mixins directory (src/assets/styles/mixins)?

File: src/index.html.tpl [hunk 3]
```diff
@@ -20,4 +20,4 @@
 <script src="<%- appJsPath %>"></script>
 <%= liveReload === true ? '<script src="//localhost:35729/livereload.js"></script>' : '' %>
 </body>
-</html>
+</html>
```
Comment (jleft) [inline src/index.html.tpl]: Please add the new line at the end of the file back in.

File: src/assets/styles/mixins.less [hunk 4]
```diff
@@ -1 +1,30 @@
 @import "mixins/buttons.less";
+
+// Device Configurations
```
Comment (jleft) [inline src/assets/styles/mixins.less]: Please can these mixins be moved into their own file? _mixins/responsive-display.less_

File: src/assets/styles/variables.less [hunk 5]
```diff
@@ -160,3 +160,33 @@
 @spinner-size: 50px;
 @spinner-thickness: 7px;
 @spinner-animation-duration: 0.6s;
+
+// If the period dropdown is shown, the period tab and clear all should be hidden
+// Phone
+@phone-landscape-nav: @hide;
```
Comment (jleft) [inline src/assets/styles/variables.less]: I'd add `-display` as a suffix to all of these variables.

**Comment by jleft** [comment]
Thanks @rjmcneill

**Pull request merged**